src/serac/physics/utilities/variational_form/integral.hpp

```diff
@@ -121,30 +170,33 @@ auto Preprocess(T u, const tensor<double, geometry_dim> xi,
   }
 }
 
-// the impl::Postprocess function computes residual contributions from the output of the q-function
-//
-// this involves integrating the q-function output against functions from the test
-// function space.
-//
-// by default:
-//   H1 family elements integrate std::get<0>(f) against the test space shape functions
-//                            and std::get<1>(f) against the test space shape function gradients
-//
-//   Hcurl family elements integrate std::get<0>(f) against the test space shape functions
-//                               and std::get<1>(f) against the curl of the test space shape functions
-//
-//   TODO:
-//   Hdiv family elements integrate std::get<0>(f) against the test space shape functions
-//                              and std::get<1>(f) against the divergence of the test space shape functions
-//
-//   TODO:
-//   L2 family elements integrate f against test space shape functions
-//
-// in the future, the user will be able to override these defaults
-// to omit unused components (e.g. provide only the term to be integrated against test function gradients)
+/**
+ * @brief Computes residual contributions from the output of the q-function
+ * This involves integrating the q-function output against functions from the
+ * test function space.
+ *
+ * By default:
+ *  H1 family elements integrate std::get<0>(f) against the test space shape functions
+ *                           and std::get<1>(f) against the test space shape function gradients
+ *  Hcurl family elements integrate std::get<0>(f) against the test space shape functions
+ *                              and std::get<1>(f) against the curl of the test space shape functions
+ * TODO: Hdiv family elements integrate std::get<0>(f) against the test space shape functions
+ *                                  and std::get<1>(f) against the divergence of the test space shape functions
+ * TODO: L2 family elements integrate f against test space shape functions
+ *
+ * In the future, the user will be able to override these defaults
+ * to omit unused components (e.g. provide only the term to be integrated against test function gradients)
+ * @tparam element_type The type of the element (used to determine the family)
+ * @tparam T The type of the output from the user-provided q-function
+ * @pre T must be a pair type for H1, H(curl) and H(div) family elements
+ * @param[in] f The value component output of the user's quadrature function (as opposed to the value/derivative pair)
+ * @param[in] xi The position of the quadrature point in reference space
+ * @param[in] J The Jacobian of the element transformation at the quadrature point
+ */
 template <typename element_type, typename T, int dim>
 auto Postprocess(T f, const tensor<double, dim> xi, const tensor<double, dim, dim> J)
 {
+  // TODO: Helpful static_assert about f being tuple or tuple-like for H1, hcurl, hdiv
```

Comment:
Later on, the type `T` will not always be a tuple here (e.g. once we support gradient-only annotations). But in the near future, we can have
```suggestion
static_assert(is_tuple<T>::value, "impl::Postprocess error: first argument must be a std::tuple of two entries (see equation (1) here https://libceed.readthedocs.io/en/latest/libCEEDapi/)");
```